### PR TITLE
Add Organise Command and Modify UI

### PR DESCRIPTION
**Pull request by zichen-3974**
Changes:
- Added organise command, user can now organise based on recommendation index or customise a meeting by providing day(d/), location(l/) and time(T/)
- Modify UI for recommendation list and organised meet up list

To fix:

- [x] Abstraction for long methods in organise command and parser
- [x] New handler for meet up index
- [x] New meet up index type

### CODE REVIEW

**Review by russelltankaimin — CHANGES_REQUESTED** [review]
Good job so far! Must be challenging so far! 
Few things:
1. Some of your methods are too long. You should consider trying to abstract out different processes in `execute` and `parse`.
2. You really should consider having a new Index. I can help you with that. Do text me about it. This is because, logically it does not make sense for a `MeetUpIndex` to suddenly be of type `ContactIndex`. It may potentially lead to misunderstanding in the future amongst developers. Try to abstract logic as much as possible here. If both are the same thing, then perhaps `ContactIndex` and your `MeetUpIndex` should inherit from an abstract class `QueryIndex`. Then you can use generics and polymorphism on `IndexHandler`

File: src/main/java/seedu/address/logic/commands/OrganiseCommand.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        try {
+            model.addMeetUp(meetUp);
+        } catch (DuplicateMeetUpException dm) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+
+        model.updateObservableMeetUpList();
+        return new CommandResult(MESSAGE_ORGANISE_NEW_MEETING_SUCCESS);
+    }
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/logic/commands/OrganiseCommand.java:120]: Agree too, will abstract out. Thanks!

File: src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        //for participants
+        List<String> indexArray = Arrays.stream(argumentMultimap.getPreamble()
+                .split(" "))
+                .filter(x -> !x.isEmpty())
+                .collect(Collectors.toList());
+        Set<ContactIndex> indices = ParserUtil.parseIndices(indexArray);
+        if (indices.isEmpty()) {
+            throw new ParseException(MESSAGE_NO_PARTICIPANTS_GIVEN);
+        }
+        Participants participants = new Participants(indices);
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java:68]: Will do. Thanks!

File: src/main/java/seedu/address/logic/parser/IndexHandler.java [hunk 3]
```diff
@@ -48,6 +49,30 @@ public Optional<Person> getPersonByIndex(ContactIndex index) {
...
+                                .getMeetUpIndex().equals(new ContactIndex(integer)))).max();
+        Integer availableIndex = takenIndices.getAsInt() + 1;
+        return new ContactIndex(availableIndex);
+    }
+
+    public Optional<MeetUp> getMeetUpByIndex(ContactIndex index) {
+        List<MeetUp> meetUpList = model.getObservableMeetUpList();
+        return meetUpList.stream()
+                .filter(meetUp -> meetUp.getMeetUpIndex().equals(index)).findFirst();
+    }
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/logic/parser/IndexHandler.java:74]: Agree, will do. Thanks!

File: src/main/java/seedu/address/model/meetup/Participants.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.model.person.ContactIndex;
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a group of participants in a meetup by their indexes.
+ */
+public class Participants {
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/model/meetup/Participants.java:13]: Thanks! :)

File: src/main/java/seedu/address/model/meetup/UniqueMeetUpList.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        if (index == -1) {
+            throw new MeetUpNotFoundException();
+        }
+
+        if (!target.isSameMeetUp(editedMeetUp) && contains(editedMeetUp)) {
+            throw new DuplicateMeetUpException();
+        }
+
+        internalList.set(index, editedMeetUp);
+    }
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/model/meetup/UniqueMeetUpList.java:63]: Thanks! :]

File: src/main/java/seedu/address/model/meetup/MeetUp.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    private final TimePeriod timePeriod;
+    private final Location location;
+    private final Participants participants;
+    private final ContactIndex meetUpIndex;
+
+    /**
+     * Constructor for a {@code MeetUp} object.
+     */
+    public MeetUp(Recommendation recommendation, Participants participants, ContactIndex meetUpIndex) {
+        this.timePeriod = recommendation.getTimePeriod();
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/model/meetup/MeetUp.java:21]: Yes agree, will create new handler. Thanks!

File: src/main/java/seedu/address/logic/commands/OrganiseCommand.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+            Recommendation recommendation = model.getRecommendationByIndex(this.index).get();
+            Participants participants = model.getParticipants();
+
+            for (Person person : participants.getParticipants()) {
+                if (!model.hasPerson(person)) { //meet -> delete someone -> organise
+                    throw new CommandException(String.format(MESSAGE_NO_SUCH_PERSON, person.getName()));
+                }
+            }
+
+            ContactIndex newIndex = model.getMeetUpIndex();
```
[the remark this answers is not in the document]
Comment (zichen-3974) [reply src/main/java/seedu/address/logic/commands/OrganiseCommand.java:89]: Yeah definitely sounds weird, will create new index for meetups.

File: src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+     */
+    private TimePeriod parseTimePeriod(ArgumentMultimap argumentMultimap) throws ParseException {
+        List<String> time = List.of(argumentMultimap.getValue(Prefix.TIME).get().split(" "));
+        LocalTime startHour;
+        LocalTime endHour;
+        try {
+            startHour = new LocalTime(Integer.parseInt(time.get(0)), 0);
+            endHour = new LocalTime(Integer.parseInt(time.get(1)), 0);
+        } catch (NumberFormatException | IllegalFieldValueException nfe) {
+            throw new ParseException(MESSAGE_WRONG_TIME_FORMAT);
```
Comment (russelltankaimin) [inline src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java:130]: FYI, can use ParserUtil for this. Helps you validate input also.

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 9]
```diff
@@ -243,6 +248,11 @@ public void setUser(User user) {
             throw new AssertionError("This method should not be called.");
         }
 
+        @Override
+        public Optional<Person> getPersonByIndex(ContactIndex index) {
+            return Optional.empty();
+        }
+
```
Comment (seanfirefox) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:255]: Remember to change to 
```
@Override
public Optional<Person> getPersonByIndex(ContactIndex index) {
        throw new AssertionError("This method should not be called.");
}
```

File: src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java (hunk 8 above)
Comment (russelltankaimin) [reply src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java:130]: We keep this for v1.4 yah?

**Review by russelltankaimin — APPROVED** [review]
Ok Code looks much better. I will merge and test it in the morning.

File: src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java (hunk 8 above)
Comment (russelltankaimin) [reply src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java:130]:
```java
public static LocalTime parseStartHour(int hour) throws ParseException```

and 
```java
public static LocalTime parseEndHour(int hour) throws ParseException
```

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java (hunk 9 above)
Comment (zichen-3974) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:255]: Ok thanks!

File: src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java (hunk 8 above)
Comment (zichen-3974) [reply src/main/java/seedu/address/logic/parser/OrganiseCommandParser.java]: Alright, changed!

**Review by russelltankaimin — APPROVED** [review]
Ok. Thank You. LGTM!

**Pull request merged**